File: BE/Key-Hub-Api/src/main/java/com/example/Keyhub/data/entity/Blog/Blog.java
package com.example.Keyhub.data.entity.Blog;

import com.example.Keyhub.data.entity.ProdfileUser.Users;
import com.example.Keyhub.data.entity.Blog.Series;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigInteger;
import java.util.Date;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "blog")
public class Blog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "blog_id")
    private BigInteger id;
    private String title;
    @Column(columnDefinition = "TEXT")
    private String content;

    private String description;
    @ManyToMany
    @JoinTable(name = "blog_tag",
            joinColumns = @JoinColumn(name = "blog_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id"))
    private List<Tag> tags;
    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
    @ManyToOne
    @JoinColumn(name = "user_id")
    private Users user;
    @Column
    private Date create_date;
    @Column
    private int status;
    @Column
    private String avatar;
    @ManyToOne
    @JoinColumn(name = "series_id")
    private Series series;
    @Column
    private BigInteger likes;
    @Column
    private Long Views;
}

File: FE/src/types/blog.ts
import User from "./user";

interface BlogPost {
  id: number;
  title: string;
  content: string;
  description: string;
  views?: number | null;
  create_date?: string;
  avatar: string;
  status_id?: number;
  likes?: number;
  categories: Category;
  tags: Tag[];
  series: BlogSeries; // Nếu series có thể là một object thì bạn cần định nghĩa một interface riêng cho nó
  isLike?: boolean;
  isSave?: boolean;
  users: User;
  sumComment?: number;
}

interface Category {
  id: number;
  name: string;
}

interface Tag {
  id: number;
  name: string;
}

interface BlogSeries {
  id: number;
  name: string;
  sumBlog: number;
  description: string;
  createday: string;
}

export default BlogPost;
